src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/IconTintColor/IconTintColorBehavior.windows.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Numerics;
+using Microsoft.Maui.Platform;
+using Microsoft.UI.Composition;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Hosting;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Media.Imaging;
+using WButton = Microsoft.UI.Xaml.Controls.Button;
+using WImage = Microsoft.UI.Xaml.Controls.Image;
+
+namespace CommunityToolkit.Maui.Behaviors;
+
+public partial class IconTintColorBehavior
+{
+	SpriteVisual? spriteVisual;
+	Vector2? originalImageSize;
+	bool IsUpdate => originalImageSize is not null;
+
+	/// <inheritdoc/>
+	protected override void OnAttachedTo(View bindable, FrameworkElement platformView)
+	{
+		ApplyTintColor(platformView, bindable, TintColor);
+
+		bindable.PropertyChanged += OnElementPropertyChanged;
+		this.PropertyChanged += (s, e) =>
+		{
+			if (e.PropertyName == TintColorProperty.PropertyName)
+			{
+				ApplyTintColor(platformView, bindable, TintColor);
+			}
+		};
+	}
+
+	/// <inheritdoc/>
+	protected override void OnDetachedFrom(View bindable, FrameworkElement platformView)
+	{
+		bindable.PropertyChanged -= OnElementPropertyChanged;
+		RemoveTintColor(platformView);
+		originalImageSize = null;
+	}
+
+	static bool TryGetButtonImage(WButton button, [NotNullWhen(true)] out WImage? image)
+	{
+		image = button.Content as WImage;
+		return image is not null;
+	}
+
+	static bool TryGetSourceImageUri(WImage? imageControl, IImageElement? imageElement, [NotNullWhen(true)] out Uri? uri)
+	{
+		if (imageElement?.Source is UriImageSource uriImageSource)
+		{
+			uri = uriImageSource.Uri;
+			return true;
+		}
+
+		if (imageControl?.Source is BitmapImage bitmapImage)
+		{
+			uri = bitmapImage.UriSource;
+			return true;
+		}
+
+		uri = null;
+		return false;
+	}
+
+	void OnElementPropertyChanged(object? sender, PropertyChangedEventArgs e)
+	{
+		if (e.PropertyName is not string propertyName
+			|| sender is not View bindable
+			|| bindable.Handler?.PlatformView is not FrameworkElement platformView)
+		{
+			return;
+		}
+
+		if (propertyName.Equals(Image.SourceProperty.PropertyName, StringComparison.Ordinal)
+			|| propertyName.Equals(ImageButton.SourceProperty.PropertyName, StringComparison.Ordinal))
+		{
+			ApplyTintColor(platformView, bindable, TintColor);
+		}
+	}
+
+	void ApplyTintColor(FrameworkElement platformView, View element, Color? color)
+	{
+		if (color is null)
+		{
+			RemoveTintColor(platformView);
+		}
```

Comment:
What happens if the user passes in `Colors.Transparent`? 

Should we also call `RemoveTintColor()`?

Reply:
I reworked that part so currently its not required to handle this case explicitly.  Technically we could still add a check and skip the composition part what might be beneficial for performance/memory usage. What do you think? @brminnick